bin/configs/kotlin-spring-boot-delegate.yaml

```diff
@@ -8,5 +8,6 @@ additionalProperties:
   annotationLibrary: swagger2
   useSwaggerUI: "true"
   delegatePattern: "true"
+  appendRequestToHandler: "true"
```

Comment:
i set this option to false but i got compilation error in `samples/server/petstore/kotlin-springboot-delegate`

does it work for you locally if you set the option `appendRequestToHandler` to false?

Reply:
Looks like the build.gradle.kts is broken. 
Do you get this error?

```

    ~/git/openapi-generator/samples/server/petstore/kotlin-springboot-delegate    issue_18555 +4 !1  gradle build                                                                                                                                                                                        ✔ 

> Task :compileKotlin FAILED
e: /home/peter/git/openapi-generator/samples/server/petstore/kotlin-springboot-delegate/src/main/kotlin/org/openapitools/api/PetApiController.kt: (7, 19): Unresolved reference: Generated
e: /home/peter/git/openapi-generator/samples/server/petstore/kotlin-springboot-delegate/src/main/kotlin/org/openapitools/api/PetApiDelegate.kt: (18, 19): Unresolved reference: Generated
e: /home/peter/git/openapi-generator/samples/server/petstore/kotlin-springboot-delegate/src/main/kotlin/org/openapitools/api/StoreApiController.kt: (7, 19): Unresolved reference: Generated
e: /home/peter/git/openapi-generator/samples/server/petstore/kotlin-springboot-delegate/src/main/kotlin/org/openapitools/api/StoreApiDelegate.kt: (17, 19): Unresolved reference: Generated
e: /home/peter/git/openapi-generator/samples/server/petstore/kotlin-springboot-delegate/src/main/kotlin/org/openapitools/api/UserApiController.kt: (7, 19): Unresolved reference: Generated
e: /home/peter/git/openapi-generator/samples/server/petstore/kotlin-springboot-delegate/src/main/kotlin/org/openapitools/api/UserApiDelegate.kt: (17, 19): Unresolved reference: Generated

FAILURE: Build failed with an exception.

```

I get it regardless of the `appendRequestToHandler` setting. It looks like the mustache template for the spring-boot < 3 is broken. Here https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/kotlin-spring/libraries/spring-boot/buildGradleKts.mustache#L60 and the next line it needs javax instead of jakarta.

These two lines should be:

```
    compile("javax.validation:validation-api")
    compile("javax.annotation:javax.annotation-api:1.3.2")
```

I can create a PR for this.

Reply:
But why didn't the workflows fail?